File: app/src/main/java/com/mmutert/freshfreezer/data/converters/OffsetUnitConverter.java
package com.mmutert.freshfreezer.data.converters;

import androidx.room.TypeConverter;

import com.mmutert.freshfreezer.data.TimeOffsetUnit;


public class OffsetUnitConverter {

    @TypeConverter
    public static String fromOffsetUnit(TimeOffsetUnit unit) {
        return unit.toString();
    }

    @TypeConverter
    public static TimeOffsetUnit toOffsetUnit(String unit) {
        return TimeOffsetUnit.valueOf(unit);
    }
}

File: app/src/main/java/com/mmutert/freshfreezer/data/converters/OffsetUnitConverter.kt
package com.mmutert.freshfreezer.data.converters

import androidx.room.TypeConverter
import com.mmutert.freshfreezer.data.TimeOffsetUnit

object OffsetUnitConverter {
    @JvmStatic
    @TypeConverter
    fun fromOffsetUnit(unit: TimeOffsetUnit): String {
        return unit.toString()
    }

    @JvmStatic
    @TypeConverter
    fun toOffsetUnit(unit: String?): TimeOffsetUnit {
        return TimeOffsetUnit.valueOf(unit!!)
    }
}